server/src/main/java/org/tctalent/server/service/db/impl/CandidateAttachmentsServiceImpl.java

```diff
@@ -145,15 +145,11 @@ public CandidateAttachment createCandidateAttachment(CreateCandidateAttachmentRe
         Candidate candidate;
         String textExtract;
 
-        // Handle requests coming from the admin portal
         if (request.getCandidateId() != null) {
             candidate = candidateRepository.findById(request.getCandidateId())
                     .orElseThrow(() -> new NoSuchObjectException(Candidate.class, request.getCandidateId()));
         } else {
-            candidate = authService.getLoggedInCandidate();
-            if (candidate == null) {
-                throw new InvalidSessionException("Not logged in");
-            }
+            throw new InvalidRequestException("Missing candidate ID");
```

Comment:
I'm not sure about this. The original code has 2 branches. The first, where candidate ID is null, is for the admin portal; the other, I presume, must be for the candidate portal.

If so, I think we need to leave the second branch intact but go to the candidate service instead of the auth service.

So, following the pattern in all your other changes, the code would become:

```
        // Handle requests coming from the admin portal
        if (request.getCandidateId() != null) {
            candidate = candidateRepository.findById(request.getCandidateId())
                    .orElseThrow(() -> new NoSuchObjectException(Candidate.class, request.getCandidateId()));
        } else {
            candidate = candidateService.getLoggedInCandidate()
                    .orElseThrow(() -> new InvalidSessionException("Not logged in"));
        }

```

I hope I'm understanding that right. Can you take one more look before we hotfix this?

Reply:
Hi @sadatmalik - in this particular example, a candidateID is always present whether it comes from the candidate portal or the admin portal. However it did used to come seperately hence the two if/else statements for getting a candidate. We used to call the createCandidateAttachment straight from the portal api create() method (and this request didn't contain the candidateID), however that is no longer used and zombie. Instead we call the upload method from the portal api, and then in that upload method we create the request and pass in the candidate Id. So that is why this method always has a candidateId.

Should I delete any zombie code I find in this hotfix PR? Or safest to create a new issue?

Reply:
Thanks @cazcam34 , best to leave any further cleanup for a separate PR pls. With your above comments, I'm happy with this to go as a Hotfix.